### ISSUE TITLE
[Issue] Image with "--" in name causes "invalid reference" error

### ISSUE THREAD

**Issue by AnrichVS** [regclient/regclient#494]
### Current Behavior

Images with "--" in their name are considered invalid references by RegCtl. 

### Expected Behavior

In image with a "--" in the name is a valid reference. 

### Steps To Reproduce

This simple example can be used:

1. `docker pull hello-world`
2. `docker tag hello-world:latest some-registry.com/project/hello--world:latest`
3. `docker push some-registry.com/project/hello--world:latest`

**Important**: note that the image name now has "--" in the name: "hello--world". 

Now attempt to delete the tag:
```shell
docker container run --rm regclient/regctl:latest  tag delete "docker push some-registry.com/project/hello--world:latest"
```

The following error is returned:
```shell
invalid reference "docker push some-registry.com/project/hello--world:latest"
```

If I push the image as "some-registry.com/project/hello-world:latest" (single dash), RegCtl correctly deletes the tag. 

### Version

```text
VCSTag:     v0.5.0
VCSRef:     c7e747e56eb507aa7f44e61362a3bd9f8b0b37fb-dirty
VCSCommit:  c7e747e56eb507aa7f44e61362a3bd9f8b0b37fb
VCSState:   dirty
VCSDate:    2023-06-28T17:37:27Z
Platform:   linux/amd64
GoVer:      go1.20.5
GoCompiler: gc
```

### Environment

- Running as binary or container: Container ([latest](https://hub.docker.com/layers/regclient/regctl/latest/images/sha256-14bbe30867b1f36924d2497f5af4d46122183670e33986f9f7334063923a9519?context=explore))
- Host platform: Arch Linux
- Registry description: [Harbor](https://goharbor.io/)

**Comment by sudo-bmitch** [comment]
This was fixed in #489. You can test the most recent changes to main with the edge tag:

```
docker container run --rm regclient/regctl:edge  tag delete "docker push some-registry.com/project/hello--world:latest"
```

**Comment by AnrichVS** [comment]
Perfect, thank you for the feedback.

I can confirm that the issue is not present on `regclient/regctl:edge`. 

It is also not present on `regclient/regctl:v0.4`, so I will use that for now until the fix reaches `regclient/regctl:latest`.
